src/groupeddatatable/grouping.jl

```diff
@@ -355,16 +355,23 @@ dt |> groupby(:a) |> [sum, x->mean(dropnull(x))]   # equivalent
 """
 aggregate(d::AbstractDataTable, fs::Function; sort::Bool=false) = aggregate(d, [fs], sort=sort)
 function aggregate{T<:Function}(d::AbstractDataTable, fs::Vector{T}; sort::Bool=false)
-    headers = _makeheaders(fs, _names(d))
-    _aggregate(d, fs, headers, sort)
+    headers = [Symbol(c, "_", f) for f in fs for c in names(d)]
+    res = DataTable(Any[f(d[c]) for f in fs for c in names(d)], headers)
+    sort && sort!(res, cols=headers)
+    res
 end
 
 # Applies aggregate to non-key cols of each SubDataTable of a GroupedDataTable
 aggregate(gd::GroupedDataTable, f::Function; sort::Bool=false) = aggregate(gd, [f], sort=sort)
 function aggregate{T<:Function}(gd::GroupedDataTable, fs::Vector{T}; sort::Bool=false)
-    headers = _makeheaders(fs, _setdiff(_names(gd), gd.cols))
-    res = combine(map(x -> _aggregate(without(x, gd.cols), fs, headers), gd))
-    sort && sort!(res, cols=headers)
+    res = gd.parent[gd.idx[gd.starts], gd.cols]
+    cols = setdiff(names(gd.parent), gd.cols)
+    for f in fs
+        for c in cols
+            res[Symbol(c, "_", f)] = [f(g[c]) for g in gd]
```

Comment:
This doesn't seem to support the case where `f` returns a vector. Though that was already the case of the previous implementation (which stored the returned vector in a cell). I guess we should just remove the mention of vectors from the docstring.